src/self.h

```diff
@@ -45,17 +47,40 @@ typedef struct {
 typedef struct {
 	uint32_t type;
 	uint32_t size;
-	uint32_t unk;
+	uint32_t has_next;
 	uint32_t pad;
 } SCE_controlinfo;
 
+
+typedef enum {
+	SCE_ELF_DIGEST_INFO = 4,
+	SCE_NPDRM_INFO = 5,
+	SCE_BOOTPARAM_INFO = 6,
+	SCE_SHAREDSECRET_INFO = 7,
+} SCE_controlinfo_type;
+
 typedef struct {
-	SCE_controlinfo common;
-	char unk[0x100];
-} SCE_controlinfo_5;
+	SCE_controlinfo head;
+	uint8_t constant[0x14];   /* same for every PSVita/PS3 SELF, hardcoded as: 627CB1808AB938E32C8C091708726A579E2586E4 */
+	uint8_t elf_digest[0x20]; /* on PSVita: SHA-256 of source ELF file, on PS3: SHA-1 */
+	uint8_t padding[8];
+	uint32_t min_required_fw; /* ex: 0x363 for 3.63, seems to be ignored */
+} SCE_elf_digest_info;
 
 typedef struct {
-	SCE_controlinfo common;
+	SCE_controlinfo head;
+	uint32_t magic;               /* 7F 44 52 4D (".DRM") */
+	uint32_t finalized_flag;      /* ex: 80 00 00 01 */
+	uint32_t drm_type;            /* license_type ex: 2 local, 0xD free with license */
+	uint32_t padding;
+	uint8_t content_id[0x30];
+	uint8_t digest[0x10];         /* ?sha-1 hash of debug self/sprx created using make_fself_npdrm? */
+	uint8_t padding_78[0x78];
+	uint8_t hash_signature[0x38]; /* unknown hash/signature */
```

Comment:
a code doesn't have to set variables.
for example in this document, drm_type only can set `2` or `0xD`, but we pass the `0`

can you update a document about behaviors when all values are 0?
current code could be same as before, so maybe we need `TODO` tag in code or need to update the document to prevent misunderstanding.

Reply:
Yeah I can add a comment for that, I didn't add code to set it because that wasn't set before so I didn't want to accidentally break something by adding that